src/MissionManager/TransectStyleComplexItem.cc

```diff
@@ -51,7 +51,7 @@ TransectStyleComplexItem::TransectStyleComplexItem(PlanMasterController* masterC
     , _terrainAdjustMaxClimbRateFact    (settingsGroup, _metaDataMap[terrainAdjustMaxClimbRateName])
     , _terrainAdjustMaxDescentRateFact  (settingsGroup, _metaDataMap[terrainAdjustMaxDescentRateName])
 {
-    _terrainQueryTimer.setInterval(_terrainQueryTimeoutMsecs);
+    _terrainQueryTimer.setInterval(qgcApp()->runningUnitTests() ? 0 : _terrainQueryTimeoutMsecs);
```

Comment:
So this timer is used to batch up repeated calls to query terrain as well as de-dup calls. This can happen from an end user futzing with things in the ui as well as the internal code itself not being super careful about asking for the same thing more than once. The first case can't happen in a unit test, but the second case of internal code asking for the same thing multiple times can. So in order to replicate what actually happens in the real world the unit test timeout should be something like say 10 msecs. That way the test past ends up going all the way out to the main loop and then back when the timer fires. Which in turn replicates what happen in the normal non unit test code path.

Reply:
good point, done: https://github.com/mavlink/qgroundcontrol/pull/8846/commits/50180bf88d353a04bd70df71f92d8c21c005dd0e